app/lib/package/model_properties.dart

```diff
@@ -135,6 +135,13 @@ class Pubspec {
               .intersect(VersionConstraint.parse('>=2.0.0'))
               .isEmpty);
 
+  late final isDart3Incompatible =
+      !supportsOnlyLegacySdk && // do not mix is:legacy with is:dart3-incompatible
+          _hasDartSdkConstraint &&
+          !_dartSdkConstraint!
+              .intersect(VersionConstraint.parse('<2.12.0'))
```

Comment:
I'm guessing this is slightly better:
```suggestion
              .intersect(VersionConstraint.parse('<2.12.0-0'))
```